### PR TITLE
Drop dependency on NamedArrays package

### LINKED ISSUE #389: Drop named argument to ranef now that we have raneftables
If we drop the `named` argument to `ranef`, now that we have `raneftables`, we can remove the dependency on the `NamedArrays` package.  Shall we do that?

### PR DESCRIPTION
**Pull request by dmbates**
- drop the `named` argument to `ranef` and the dependency on the `NamedArrays` package that it required
- the `raneftables` function provides an alternative that, in most cases, is easier to work with.
- closes #389

### CODE REVIEW

File: src/bootstrap.jl [hunk 1]
```diff
@@ -15,7 +15,7 @@ Tables.Schema:
  :σ          T
  :β          NamedTuple{β_names}{NTuple{p,T}}
  :se         StaticArrays.SArray{Tuple{p},T,1,p}
- :θ          StaticArrays.SArray{Tuple{p},T,1,k}
+ :θ          StaticArrays.SArray{Tuple{k},T,1,k}
```
Comment (palday) [inline src/bootstrap.jl:18]: This was a copy-paste error I introduced, right?

**Review by palday — APPROVED** [review]

File: src/bootstrap.jl (hunk 1 above)
Comment (dmbates) [reply src/bootstrap.jl:18]: Yes.

Sorry for putting that in this PR without explanation.  I was trying to decide if the `StaticArrays` package was still needed and found this while doing a search on the name.

Comment (palday) [reply src/bootstrap.jl:18]: I thought as much, no worries!

**Comment by palday** [comment]
I'm happy to squash and merge when you're ready.

**Pull request merged**